packages/config/src/headers.js

```diff
@@ -19,33 +19,30 @@ const HEADERS_FILENAME = '_headers'
 const addConfigHeaders = async function ({ headers: configHeaders = [], ...config }, headersPath, logs) {
   try {
     const normalizedConfigHeaders = normalizeHeaders(configHeaders)
-    const normalizedFileHeaders = await getFileHeaders(headersPath, normalizedConfigHeaders)
+    const normalizedFileHeaders = await getFileHeaders(headersPath)
     const headers = mergeHeaders({ fileHeaders: normalizedFileHeaders, configHeaders: normalizedConfigHeaders })
-    return { ...config, headers }
+    const uniqueHeaders = headers.filter(isUniqueHeader)
```

Comment:
In a follow-up PR, I will move this logic to `mergeHeaders()` and `mergeRedirects()` instead.